pkgs/development/tools/build-managers/meson/rpath.patch

```diff
@@ -1,12 +1,353 @@
+-        self.sh_name = struct.unpack(self.Word, ifile.read(self.WordSize))[0]
+-# Elf64_Word
+-        self.sh_type = struct.unpack(self.Word, ifile.read(self.WordSize))[0]
+-# Elf64_Xword
+-        if is_64:
+-            self.sh_flags = struct.unpack(self.XWord, ifile.read(self.XWordSize))[0]
+-        else:
+-            self.sh_flags = struct.unpack(self.Word, ifile.read(self.WordSize))[0]
+-# Elf64_Addr
+-        self.sh_addr = struct.unpack(self.Addr, ifile.read(self.AddrSize))[0]
+-# Elf64_Off
+-        self.sh_offset = struct.unpack(self.Off, ifile.read(self.OffSize))[0]
+-# Elf64_Xword
+-        if is_64:
+-            self.sh_size = struct.unpack(self.XWord, ifile.read(self.XWordSize))[0]
+-        else:
+-            self.sh_size = struct.unpack(self.Word, ifile.read(self.WordSize))[0]
+-# Elf64_Word
+-        self.sh_link = struct.unpack(self.Word, ifile.read(self.WordSize))[0]
+-# Elf64_Word
+-        self.sh_info = struct.unpack(self.Word, ifile.read(self.WordSize))[0]
+-# Elf64_Xword
+-        if is_64:
+-            self.sh_addralign = struct.unpack(self.XWord, ifile.read(self.XWordSize))[0]
+-        else:
+-            self.sh_addralign = struct.unpack(self.Word, ifile.read(self.WordSize))[0]
+-# Elf64_Xword
+-        if is_64:
+-            self.sh_entsize = struct.unpack(self.XWord, ifile.read(self.XWordSize))[0]
+-        else:
+-            self.sh_entsize = struct.unpack(self.Word, ifile.read(self.WordSize))[0]
+-
+-class Elf(DataSizes):
++class Elf:
+     def __init__(self, bfile, verbose=True):
+         self.bfile = bfile
+         self.verbose = verbose
+-        self.bf = open(bfile, 'r+b')
+-        try:
+-            (self.ptrsize, self.is_le) = self.detect_elf_type()
+-            super().__init__(self.ptrsize, self.is_le)
+-            self.parse_header()
+-            self.parse_sections()
+-            self.parse_dynamic()
+-        except:
+-            self.bf.close()
+-            raise
+-
+-    def __enter__(self):
+-        return self
+-
+-    def __del__(self):
+-        if self.bf:
+-            self.bf.close()
+-
+-    def __exit__(self, exc_type, exc_value, traceback):
+-        self.bf.close()
+-        self.bf = None
+-
+-    def detect_elf_type(self):
+-        data = self.bf.read(6)
+-        if data[1:4] != b'ELF':
+-            # This script gets called to non-elf targets too
+-            # so just ignore them.
+-            if self.verbose:
+-                print('File "%s" is not an ELF file.' % self.bfile)
+-            sys.exit(0)
+-        if data[4] == 1:
+-            ptrsize = 32
+-        elif data[4] == 2:
+-            ptrsize = 64
+-        else:
+-            sys.exit('File "%s" has unknown ELF class.' % self.bfile)
+-        if data[5] == 1:
+-            is_le = True
+-        elif data[5] == 2:
+-            is_le = False
+-        else:
+-            sys.exit('File "%s" has unknown ELF endianness.' % self.bfile)
+-        return ptrsize, is_le
+-
+-    def parse_header(self):
```

Comment:
When it goes into the tree, I'd suggest making the patch as small as possible so that it's not a maintenance burden. You probably already have that in mind, but just in case...

Since Python functions can be redefined, I'd suggest storing patchelf versions of `self.print_soname()` and `self.fix_rpath()` in a `.py` file that then will be appended to `scripts/depfixer.py` in `patchPhase`.